azurerm/resource_arm_dns_ns_record.go

```diff
@@ -152,35 +171,64 @@ func resourceArmDnsNsRecordDelete(d *schema.ResourceData, meta interface{}) erro
 	return nil
 }
 
-func flattenAzureRmDnsNsRecords(records *[]dns.NsRecord) []map[string]interface{} {
+//TODO: remove this once we remove the `record` attribute
+func flattenAzureRmDnsNsRecordsSet(records *[]dns.NsRecord) []map[string]interface{} {
 	results := make([]map[string]interface{}, 0, len(*records))
 
 	if records != nil {
 		for _, record := range *records {
 			nsRecord := make(map[string]interface{})
 			nsRecord["nsdname"] = *record.Nsdname
-
 			results = append(results, nsRecord)
 		}
 	}
 
 	return results
 }
 
-func expandAzureRmDnsNsRecords(d *schema.ResourceData) ([]dns.NsRecord, error) {
-	recordStrings := d.Get("record").(*schema.Set).List()
-	records := make([]dns.NsRecord, len(recordStrings))
+func flattenAzureRmDnsNsRecords(records *[]dns.NsRecord) []string {
+	results := make([]string, 0, len(*records))
 
-	for i, v := range recordStrings {
-		record := v.(map[string]interface{})
-		nsdName := record["nsdname"].(string)
+	if records != nil {
+		for _, record := range *records {
+			results = append(results, *record.Nsdname)
+		}
+	}
+
+	return results
+}
 
-		nsRecord := dns.NsRecord{
-			Nsdname: &nsdName,
+func expandAzureRmDnsNsRecords(d *schema.ResourceData) ([]dns.NsRecord, error) {
+	var records []dns.NsRecord
+
+	//TODO: remove this once we remove the `record` attribute
+	if d.HasChange("record") {
```

Comment:
IMO we should flip this around to check for `records` then `record` (as the else statement) since `records` is now the recommended field

Reply:
Works for me. Done